meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1634 @@
+                if bone_name == "pelvis":
+                    # there's a scale mismatch somewhere and the global translation is off by a factor of 100
+                    armature.pose.bones[bone_name].location = Vector((current_trans[0]*100, current_trans[1]*100, current_trans[2]*100))
+                    armature.pose.bones[bone_name].keyframe_insert('location', frame=current_frame)
+
+                # Keyframe bone rotation
+                pose_rodrigues = current_pose[bone_index]
+                set_pose_from_rodrigues(armature, bone_name, pose_rodrigues)
+                armature.pose.bones[bone_name].keyframe_insert('rotation_quaternion', frame=current_frame)
+
+            if self.keyframe_corrective_pose_weights:
+                # Calculate corrective poseshape weights for current pose and keyframe them.
+                # Note: This significantly increases animation load time and also reduces real-time playback speed in Blender viewport.
+                bpy.ops.object.update_pose_correctives('EXEC_DEFAULT')
+                for key_block in obj.data.shape_keys.key_blocks:
+                    if key_block.name.startswith("Pose"):
+                        key_block.keyframe_insert("value", frame=current_frame)
+
+        correct_for_anim_format(self.anim_format, armature)
+
+        print(f"  {num_keyframes}/{num_keyframes}")
+        context.scene.frame_set(1)
+
+        return {'FINISHED'}
+
+
+class OP_CreateAvatar(bpy.types.Operator):
+    bl_idname = "scene.create_avatar"
+    bl_label = "Create Avatar"
+    bl_description = ("Create a SMPL family avatar at the scene origin.  \nnote: SMPLH is missing the joint regressor so you can't modify it's shape")
+    bl_options = {'REGISTER', 'UNDO'}
+
+    @classmethod
+    def poll(cls, context):
+        try:
+            # Enable button only if in Object Mode
+            SMPL_version = context.window_manager.smpl_tool.SMPL_version
+
+            if (context.active_object is None) or (context.active_object.mode == 'OBJECT'):
+                return True
+            else: 
+                return False
+        except: return False
+
+    def execute(self, context):
+        gender = context.window_manager.smpl_tool.gender
+        SMPL_version = context.window_manager.smpl_tool.SMPL_version
+
+        if (SMPL_version == 'SMPLX'):
+            # Use 300 shape model by default if available
+            model_file = SMPLX_MODELFILE
+
+        elif (SMPL_version == 'SUPR'):
+            model_file = SUPR_MODELFILE
+
+        # for now, the SMPLH option has been removed from the properties because we don't have regressors for it, 
+        # so afm and a bunch of other stuff doesn't work
+        elif (SMPL_version == "SMPLH"):
+            model_file = SMPLH_MODELFILE
+
+        else:
+            model_file = "error bad SMPL_version"
+
+        objects_path = os.path.join(PATH, "data", model_file, "Object")
+        object_name = SMPL_version + "-mesh-" + gender
```

Comment:
It reads a little cleaner to use f-strings for this kind of string building instead of adding things together:

    object_name = f'{SMPL_version}-mesh-{gender}'